doc/compatibility.md

```diff
@@ -33,23 +33,24 @@ First of all, please check the list of known issues below, [package support tabl
 
 If the issue is not known yet, please open a new issue on GitHub and include the following information:
 
-  1. Go and GopherJS versions you are using.
-  2. In which environment do you see the issue (browser, nodejs, etc.).
-  3. A minimal program that behaves differently when compiled with the regular Go compiler and GopherJS.
+1. Go and GopherJS versions you are using.
+2. In which environment do you see the issue (browser, nodejs, etc.).
+3. A minimal program that behaves differently when compiled with the regular Go compiler and GopherJS.
 
-Now that the issue exists, we (GopherJS maintainers) will do our best to address it as promptly as we can. Note, however, that all of us are working on GopherJS in our spare time after our job and family responsibilities, so we can't guarantee an immediate fix. 
+Now that the issue exists, we (GopherJS maintainers) will do our best to address it as promptly as we can. Note, however, that all of us are working on GopherJS in our spare time after our job and family responsibilities, so we can't guarantee an immediate fix.
 
 🚧 If you would like to help, please consider [submitting a pull request](https://github.com/gopherjs/gopherjs/wiki/Developer-Guidelines) with a fix. If you are unsure of the best way to approach the issue, we will be happy to share whatever knowledge we can! 😃
 
 ## How to write portable code
 
 For the most part, GopherJS shouldn't require any special support for the code that only uses [supported standard packages](packages.md).
 
-However, if you do need to provide different implementations depending on the target architecture, you can use [build constraints](https://golang.org/cmd/go/#hdr-Build_constraints) to do so:
+However, if you do need to provide different implementations depending on the target architecture, you can use [build constraints](https://golang.org/cmd/go/#hdr-Build_constraints) to do so. By default, GopherJS uses `GOOS=js GOARCH=ecmascript`, which can be used in build constraints:
 
-  - `//+build js` — the source will be used for GopherJS and Go WebAssembly, but not for native builds.
-  - `//+build js,-wasm` — the source will be used for GopherJS only, and not WebAssembly or native builds.
-  - `//+build js,wasm` — the source will be used for Go WebAssembly, and not GopherJS or native builds.
+- `//go:build js` — the source will be used for GopherJS and Go WebAssembly, but not for native builds.
+- `//go:build js && ecmascript` — the source will be used for GopherJS only, and not WebAssembly or native builds.
+- `//go:build js && wasm` — the source will be used for Go WebAssembly, and not GopherJS or native builds.
+- `//go:build gopherjs` — the source will be used only by the GopherJS compiler, regardless of the `GOOS` and `GOARCH`. Use this constraint for sources that are not portable to other Go implementations.
```

Comment:
Is this functionally equivalent to `js && ecmascript`?

Reply:
Not necessarily. Theoretically, there could be another Go to ECMAScript compiler, which isn't GopherJS (for example, [joy](https://github.com/matthewmueller/joy), though it's currently inactive). Those could use `GOOS=js` `GOARCH=ecmascript` but lack some GopherJS-specific features (e.g. support for `github.com/gopherjs/gopherjs/js` package). Different build tags give the user more control for which toolchain the particular source would be used.